crates/extensions/tedge_mqtt_ext/src/test_helpers.rs

```diff
@@ -48,14 +33,88 @@ pub async fn assert_received_includes_json<I, S>(
 {
     for expected_msg in expected.into_iter() {
         let message = messages.recv().await.expect("MQTT channel closed");
-        assert_eq!(message.topic, Topic::new_unchecked(expected_msg.0.as_ref()));
-        let payload = serde_json::from_str::<serde_json::Value>(
-            message.payload_str().expect("non UTF-8 payload"),
-        )
-        .expect("non JSON payload");
-        assert_json_include!(
-            actual: payload,
-            expected: expected_msg.1
-        );
+        assert_message_includes_json(&message, expected_msg);
+    }
+}
+
+pub fn assert_messages_contains_str<I, S>(messages: &mut Vec<MqttMessage>, expected: I)
+where
+    I: IntoIterator<Item = (S, S)>,
+    S: AsRef<str> + Debug,
+{
+    let mut expected = expected.into_iter();
+    loop {
+        match (messages.is_empty(), expected.next()) {
+            (false, Some(expected_msg)) => {
+                let message = messages.remove(0);
```

Comment:
Not too bothered about the perf of these `remove` calls (because of the shifting of elements in the vec after each call), as this is a test helper. But, if there's a better way, lemme know.